hw/hw10/hw10_sip500_2.py

```diff
@@ -0,0 +1,269 @@
+    soup = BeautifulSoup(response.content, "xml")
+    data = soup.find(ID="R01235").Value.text
+    coast = data.replace(",", ".")
+    return float(coast)
+
+
+def p_e(soup: BeautifulSoup) -> float:
+    """Parse P/E ratio from personal page of company.
+
+    Args:
+        soup: parser
+
+    Returns:
+        The return value. P/E ratio
+    """
+    string = soup.find(class_="snapshot__header", string="P/E Ratio")
+    try:
+        data = string.find_parents(class_="snapshot__data-item")
+    except AttributeError:
+        return float("inf")
+    data = data[0].text.split()[0]
+    data = data.replace(",", "")
+    return float(data)
+
+
+def code_name_price(soup: BeautifulSoup) -> Tuple[str, str, float]:
+    """Parse code, name, price shares in USD from personal page of company.
+
+    Args:
+        soup: parser
+
+    Returns:
+        The return value. Code, name, price shares in USD
+    """
+    data = soup.find("span", class_="price-section__category")
+    code = data.span.text[2:]
+    name = soup.find(class_="price-section__label").text
+    data = soup.find(class_="price-section__current-value").text
+    price_usd = float(data.replace(",", ""))
+    return code, name.strip(), price_usd
+
+
+def potential_profit(soup: BeautifulSoup) -> float:
+    """Parse week_high, week_low indicators from personal page of company.
+
+        Calculation formula: round(float(week_high) / float(week_low) * 100 - 100, 2)
+    Args:
+        soup: parser
+
+    Returns:
+        The return value. Potential profit in %
+    """
+    string = soup.find(string="52 Week Low")
+    try:
+        data = string.find_parent(
+            class_="snapshot__data-item snapshot__data-item--small"
+        )
+    except AttributeError:
+        return -float("inf")
+    week_low = list(data)[0].strip()
+    week_low = week_low.replace(",", "")
+    string = soup.find(string="52 Week High")
+    data = string.find_parent(
+        class_="snapshot__data-item snapshot__data-item--small snapshot__data-item--right"
+    )
+    week_high = list(data)[0].strip()
+    week_high = week_high.replace(",", "")
+    return round(float(week_high) / float(week_low) * 100 - 100, 2)
+
+
+async def fetch_response(url: str) -> str:
+    """Fetch response function.
+
+    Args:
+        url: url
+
+    Returns:
+        The return value. Response.text
+    """
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as response:
+            return await response.text()
+
+
+async def result(urls: List[str]) -> Future:
+    """Get the result.
+
+    Args:
+        urls: urls
+
+    Returns:
+        The return value. Future object
+    """
+    tasks = [asyncio.create_task(fetch_response(url)) for url in urls]
+    return await asyncio.gather(*tasks)
+
+
+async def pages(
+    urls: List[str], growth: Dict[str, str], rub_usd: float
+) -> Tuple[List[Dict], List[Dict], List[Dict], List[Dict]]:
```

Comment:
Here we have a massive function with 4 business purposes, it would be better to split them to different units and run them in parallel using multiprocessing if it needed for counting 10 the most_<attr_name> values